apps/aecore/src/aec_peers.erl

```diff
@@ -734,21 +746,27 @@ connect(State0) ->
             end
     end.
 
-%% Connects to a random peer from unverified pool to check if the selected peer
-%% is alive; schedule the next TCP probe timeout if required.
--spec tcp_probe(state()) -> state().
-tcp_probe(State0) ->
-    State = cancel_tcp_probe_timer(State0),
-    case pool_random_select(unverified, State) of
-        {unavailable, State2} ->
-            epoch_sync:debug("No peers available for TCP probe", []),
-            schedule_tcp_probe(?MAX_TCP_PROBE_INTERVAL, State2);
-        {wait, Delay, State2} ->
-            epoch_sync:debug("No peers available for TCP probe, "
-                             "waiting ~b ms", [Delay]),
-            schedule_tcp_probe(Delay, State2);
-        {selected, Peer, State2} ->
-            schedule_tcp_probe(undefined, start_connection(tcp, Peer, State2))
+%% Connects to a random peer from verified or the unverified pool to check if
+%% the selected peer is alive; schedule the next TCP probe timeout
+-spec tcp_probe(peer_pool_name(), state()) -> state().
+tcp_probe(PeerPoolName, State0) ->
+    State = cancel_tcp_probe_timer(PeerPoolName, State0),
+    case is_tcp_probe_allowed(State0) of
```

Comment:
Is the reference to `State0` instead of `State` deliberate?

Reply:
If so, perhaps this could be made clear with e.g.
```erlang
ProbeAllowed = is_tcp_probe_allowed(State0),
State = cancel_tcp_probe_timer(PeerPoolName, State0),
case ProbeAllowed of
   ...
end
```

Reply:
But the test doesn't seem to need to reference `State0` ...